### PR TITLE
build: Add first draft of lua build file

### PR DESCRIPTION
**Pull request by duncanthomson**
This PR adds a lua file for selectively building a subset of libgit2 components. In particular, it doesn't build the components to support ssh or https.

Please could you help me get this up to scratch?

### CODE REVIEW

File: libgit2.lua [hunk 1]
```diff
@@ -0,0 +1,477 @@
...
+  flags {
+    "NoPCH",
+  }
+
+  includedirs {
+    prjDir .. "/include/",
+    prjDir .. "/src/",
+    prjDir .. "/deps/http-parser/",
+    prjDir .. "/deps/regex/",
+    prjDir .. "/deps/zlib/"
```
Comment (chri7325) [inline libgit2.lua]: Should we use our zlib?

File: libgit2.lua [hunk 2]
```diff
@@ -0,0 +1,477 @@
...
+
+    platform_defines = {
+      "SHA1DC_CUSTOM_INCLUDE_SHA1_C=\\\"common.h\\\"",
+      "SHA1DC_CUSTOM_INCLUDE_UBC_CHECK_C=\\\"common.h\\\"",
+    }
+
+  end
+
+  defines {
+    {
```
Comment (chri7325) [inline libgit2.lua]: nitpick, extra {

File: libgit2.lua [hunk 3]
```diff
@@ -0,0 +1,477 @@
...
+      "GIT_THREADS=1",
+      "GIT_SHA1_COLLISIONDETECT=1",
+      "SHA1DC_NO_STANDARD_INCLUDES=1",
+      "LIBGIT2_NO_FEATURES_H",
+      platform_defines
```
Comment (chri7325) [inline libgit2.lua]: nitpick, missing trailing comma

File: libgit2.lua [hunk 4]
```diff
@@ -0,0 +1,477 @@
...
+    }
+  }
+
+  files {
+    "deps/http-parser/*.c",
+    "deps/regex/regex.c",
+    "deps/zlib/*.c",
```
Comment (chri7325) [inline libgit2.lua]: Should we just use our zlib?

File: libgit2.lua [hunk 5]
```diff
@@ -0,0 +1,477 @@
...
+    "src/*.c",
+    "src/hash/sha1dc/*.c",
+    "src/streams/*.c",
+    "src/transports/*.c",
+    "src/xdiff/*.c",
+    platform_files
```
Comment (chri7325) [inline libgit2.lua]: nitpick, trailing comma

File: libgit2.lua (hunk 1 above)
Comment (duncanthomson) [reply libgit2.lua]: Yes. Could you help me understand how? I don't know how to cross-reference 3rd party dependencies with our build system.

Comment (chri7325) [reply libgit2.lua]: Sure thing.  You just need to include the zlib dir as seen at https://devtopia.esri.com/3rdparty/libtiff/blob/runtimecore/libtiff.lua#L67.  Half the fun of our 3rdparty libs is that a lot of them use each other.  I try to keep the headers standalone so all needed defines are contained within the library so by adding the zlib include dir and removing the internal one, it should just work.

**Comment by duncanthomson** [comment]
Thank you @chri7325 -- please can you re-review?

**Review by dkalowsk — APPROVED** [review]

**Review by chri7325 — APPROVED** [review]

**Comment by chri7325** [comment]
Sorry about the late re-review, I was out yesterday.  Looks good!

**Pull request merged**